gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,290 @@
+      return this;
+    }
+
+    /**
+     * Removes all the records.
+     */
+    public Builder clearRecords() {
+      this.rrdatas.clear();
+      return this;
+    }
+
+    /**
+     * Replaces the current records with the provided list of records.
+     */
+    public Builder records(List<String> records) {
+      this.rrdatas = Lists.newLinkedList(checkNotNull(records));
+      return this;
+    }
+
+    /**
+     * Sets name for this DNS record set. For example, www.example.com.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the number of seconds that this record can be cached by resolvers. This number must be
+     * non-negative.
+     *
+     * @param ttl A non-negative number of seconds
+     */
+    public Builder ttl(int ttl) {
+      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was " + ttl + ".");
+      this.ttl = ttl;
+      return this;
+    }
+
+    /**
+     * The identifier of a supported record type, for example, A, AAAA, MX, TXT, and so on.
+     */
+    public Builder type(DnsRecordType type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Builds the DNS record.
+     */
+    public DnsRecord build() {
+      return new DnsRecord(this);
+    }
+  }
+
+  DnsRecord(Builder builder) {
+    this.name = builder.name;
+    this.rrdatas = ImmutableList.copyOf(builder.rrdatas);
+    this.ttl = builder.ttl;
+    this.type = builder.type;
+  }
+
+  /**
+   * Creates a builder pre-populated with the attribute values of this instance.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Creates a builder for {@code DnsRecord} with mandatorily set name and type of the record.
+   */
+  public static Builder builder(String name, DnsRecordType type) {
+    return new Builder().name(name).type(type);
+  }
+
+  /**
+   * Get the mandatory user assigned name of this DNS record.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Returns a list of DNS record stored in this record set.
+   */
+  public List<String> records() {
+    return rrdatas;
+  }
+
+  /**
+   * Returns the number of seconds that this DnsResource can be cached by resolvers.
+   */
+  public Integer ttl() {
+    return ttl;
+  }
+
+  /**
+   * Returns the type of this DNS record.
+   */
+  public DnsRecordType type() {
+    return type;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, rrdatas, ttl, type);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return (obj instanceof DnsRecord) && Objects.equals(this.toPb(), ((DnsRecord) obj).toPb());
+  }
+
+  com.google.api.services.dns.model.ResourceRecordSet toPb() {
+    com.google.api.services.dns.model.ResourceRecordSet pb =
+            new com.google.api.services.dns.model.ResourceRecordSet();
+    pb.setName(this.name());
+    pb.setRrdatas(this.records());
```

Comment:
The pb object allows it. Conceptually, this seems weird, but there may be a special meaning of a configuration with empty record set. I would allow it and let the service report an error if this is not valid in a particular situation.